### ISSUE TITLE
Setting javax.net.ssl.trustStore relative to current project does not work when building from outside project directory

### ISSUE THREAD

**Issue by ppalaga** [quarkusio/quarkus#13752]
We have a test module with the following `application.properies`:

```
quarkus.native.additional-build-args = -J-Djavax.net.ssl.trustStore=target/ssl/trust-store.jks
```

There is some rewriting of `javax.net.ssl.trustStore` in [NativeImageBuildStep.handleAdditionalProperties(NativeConfig, List<String>, boolean, Path)](https://github.com/quarkusio/quarkus/blob/bc69dfded642ab7958f800e35cd3cb4c6f113361/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java#L527-L547) when building with Docker.

This works well as long as the build is performed from the current module's directory (i.e. where the current module's pom.xml resides) or if the `javax.net.ssl.trustStore` path is set to an absolute path.

If the build is performed from outside the current project's directory - e.g. from the source tree root using `-pl`

```
mvn clean verify -Dnative -Dquarkus.native.container-build=true -pl integration-tests/solr -Dtest=foo
```

The copying fails, because the resource is not resolved relative to the current project's directory:

```
 Failed to execute goal io.quarkus:quarkus-maven-plugin:1.10.2.Final:build (quarkus-build) on project camel-quarkus-integration-test-solr: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:307)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.io.UncheckedIOException: Unable to copy trustStore file 'target/ssl/trust-store.jks' to volume root directory '/home/ppalaga/orgs/quarkus/camel-quarkus/integration-tests/solr/target/camel-quarkus-integration-test-solr-1.5.0-SNAPSHOT-native-image-source-jar'
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.handleAdditionalProperties(NativeImageBuildStep.java:498)
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:166)
[ERROR]         ... 12 more
[ERROR] Caused by: java.nio.file.NoSuchFileException: target/ssl/trust-store.jks
[ERROR]         at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
[ERROR]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
[ERROR]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
[ERROR]         at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
[ERROR]         at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)
[ERROR]         at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
[ERROR]         at java.base/java.nio.file.Files.readAttributes(Files.java:1763)
[ERROR]         at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:225)
[ERROR]         at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
[ERROR]         at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
[ERROR]         at java.base/java.nio.file.Files.walkFileTree(Files.java:2716)
[ERROR]         at io.quarkus.bootstrap.util.IoUtils.copy(IoUtils.java:126)
[ERROR]         at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.handleAdditionalProperties(NativeImageBuildStep.java:493)
```

**Expected behavior**

Building the native image should pass regardless what is the current directory.

**Actual behavior**

It fails

**To Reproduce**

```
git clone git@github.com:ppalaga/camel-quarkus.git
cd camel-quarkus
git fetch origin --tags
git reset --hard i2029-quarkus-relpath-issue

mvn clean install -Dquickly

mvn clean verify -Pnative,docker -pl integration-tests/solr -Dtest=foo
# ... and the above issue occurs
```

**Additional context**

We use `-pl` to run tests in groups in parallel on our CI.

Well apparently, we could filter the `application.properties` and set the absolute path using `${project.basedir}`. However, this is an additional complication that will make it harder to run the test in Quarkus Platform. It would be nice if building would work from any directory.

cc @zbendhiba

**Comment by zakkak** [comment]
This doesn't look like something that can be fixed in Quarkus. If I understand correctly Quarkus would need to know that the code is being run from a different directory, which seems like a maven (or gradle) responsibility.

Instead, users interested in using relative paths should probably rely on maven properties and use `${basedir}` or `${project.basedir}` as the base prefix for the relative path. Users insisting in using an application.properties file would need to [enable maven resources filtering](https://maven.apache.org/plugins/maven-resources-plugin/examples/filter.html) so they can use the same `basedir` variables in the properties file.

Something similar should be possible when using Gradle.

I am closing this as won't fix. Please re-open if you think otherwise.

**Comment by ppalaga** [comment]
> This doesn't look like something that can be fixed in Quarkus. If I understand correctly Quarkus would need to know that the code is being run from a different directory, which seems like a maven (or gradle) responsibility.

I am quite sure, it can be fixed in Quarkus. Quarkus build container is started from the Maven (or Gradle) plugin where the value of `${project.basedir}` is well known and is IIRC passed through some Quarkus APIs to most places where it is needed. 
Apparently `NativeImageBuildStep` is not using that Quarkus API and is thus not working properly when `Path.get(".")` and `${project.basedir}` are pointing to different directories.

**Comment by geoand** [comment]
Even if this is solvable, does it really make sense spending time on it? Sounds like an edge case, no?

**Comment by ppalaga** [comment]
It is maven plugins not able to handle `-f` and `-pl` CLI args that are broken, not the users' expectations.
